ampersand-select-view.js

```diff
@@ -359,15 +359,26 @@ module.exports = View.extend({
             if (!model) throw new Error('model or model idAttribute not found in options collection');
             return this.yieldModel ? model : model[this.idAttribute];
         } else if (Array.isArray(this.options)) {
-            // find value value in options array
-            // find option, formatted [['val', 'text'], ...]
-            if (this.options.length && Array.isArray(this.options[0])) {
-                for (var i = this.options.length - 1; i >= 0; i--) {
-                    if (this.options[i][0] == value) return this.options[i];
+            var i = this.options.length - 1;
+            if (i >= 0) {
+                // find value in collection as [['val','text'], ...]
+                if (Array.isArray(this.options[0])) {
+                    for (i; i >= 0; i--) {
```

Comment:
Reusing the variable `i` here seems like the wrong move, it confuses its purpose imo.

Honestly we could simplify it by scoping it (`var i;`) and then the if statement `i >= 0` can simply be `this.options.length >= 1`

Typically the `length` attribute is cast into its own var as an optimization for loops. It's only being referenced once during the if statement so it's really not necessary.